src/ship-gen/itemGeneration.js

```diff
@@ -1,26 +1,32 @@
 import ItemPrefab from "./itemPrefab";
-import Srand from "seeded-rand";
 import EntityFactories from "../entityFactories";
 import GameMap from "../gameMap"; // eslint-disable-line no-unused-vars
-import RectangularRoom from "../ship-gen/roomTypes"; // eslint-disable-line no-unused-vars
+import { RectangularRoom } from "./roomTypes"; // eslint-disable-line no-unused-vars
+import ChoiceIndex from "../utils/choiceIndex";
 
 export default class ItemGenerator {
 
     /**
      *
-     * @param {GameMap} gameMap
+     * @param {integer} difficultyLevel - The difficulty level to select the weights for (1-X)
+     * @param {GameMap} gameMap - GameMap instance
      */
-    constructor(gameMap) {
+    constructor(difficultyLevel, gameMap) {
+        this.difficultyLevel = difficultyLevel;
         this.gameMap = gameMap;
 
         this._loaded = false;
-        this._chances = [];
-        this._items = [];
-        this._sumItemWeights = 0;
+        this._itemChances = {};
+        this._items = {};
+        this._sumItemWeights = {};
+
+        this._rarityChances = [];
+        this._itemRarities = [];
+        this._sumItemRarityWeights = 0;
     }
 
     /**
-     * Spawn an item in the generators currently definied room
+     * Spawn an item in the generators currently defined room
      */
     spawnItem() {
         if (this._loaded === false) {
```

Comment:
This could just be !this._loaded

Line 36 likely needs to change to be "=== undefined" instead of "=== null"